### PR TITLE
Jwt token

### PR DESCRIPTION
**Pull request by layzimayner**

### CODE REVIEW

File: src/main/java/com/example/demo/config/OpenApiConfig.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class OpenApiConfig {
+
+    @Bean
+    public OpenAPI customOpenApi() {
+        return new OpenAPI()
+                .components(new Components().addSecuritySchemes("BearerAuth",
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/config/OpenApiConfig.java]: let's create constants for strings

File: src/main/java/com/example/demo/controller/AuthenticationController.java [hunk 2]
```diff
@@ -18,8 +21,14 @@
 @RequiredArgsConstructor
 public class AuthenticationController {
     private final UserService userService;
+    private final AuthenticationService authenticationService;
 
-    @PostMapping
+    @PostMapping("/login")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/controller/AuthenticationController.java:26]: let's add swagger annotations

File: src/main/java/com/example/demo/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                    null,
+                    userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer")) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/security/JwtAuthenticationFilter.java]: create constant for `"Bearer"`

File: src/main/java/com/example/demo/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(constant.length());
```

File: src/main/java/com/example/demo/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                    userDetails,
+                    null,
+                    userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/security/JwtAuthenticationFilter.java]: let's use Spring constant 
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some comments

File: src/main/java/com/example/demo/security/JwtAuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final String ENCODING_SCHEME = "Bearer";
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/demo/security/JwtAuthenticationFilter.java]:
```suggestion
    private static final String ENCODING_SCHEME = "Bearer ";
```

File: src/main/java/com/example/demo/security/JwtAuthenticationFilter.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+                    userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(ENCODING_SCHEME)) {
+            return bearerToken.substring(ENCODING_SCHEME.length() + 1);
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/demo/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(ENCODING_SCHEME.length());
```

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -8,4 +8,6 @@ spring.jpa.show-sql=true
 spring.liquibase.enabled=true
 spring.liquibase.change-log=classpath:db/changelog/db.changelog-master.yaml
 
+jwt.expiration=30000000
+jwt.secret=1jerjncvokjfvwlfjfvnwhjhfnv
```
Comment (liliia-ponomarenko) [inline src/main/resources/application.properties]: is it enough length of secret?

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! A few minor comments ;)

**Pull request merged**